Mathlib/RingTheory/TensorProduct.lean

```diff
@@ -594,19 +594,12 @@ tensors can be directly applied by the caller (without needing `TensorProduct.on
 def algHomOfLinearMapTensorProduct (f : A ⊗[R] B →ₗ[S] C)
     (h_mul : ∀ (a₁ a₂ : A) (b₁ b₂ : B), f ((a₁ * a₂) ⊗ₜ (b₁ * b₂)) = f (a₁ ⊗ₜ b₁) * f (a₂ ⊗ₜ b₂))
     (h_one : f (1 ⊗ₜ[R] 1) = 1) : A ⊗[R] B →ₐ[S] C :=
-  AlgHom.ofLinearMap f h_one fun x y => by
-      simp only
-      refine TensorProduct.induction_on x ?_ ?_ ?_
-      · rw [zero_mul, map_zero, zero_mul]
-      · intro a₁ b₁
-        refine TensorProduct.induction_on y ?_ ?_ ?_
-        · rw [mul_zero, map_zero, mul_zero]
-        · intro a₂ b₂
-          rw [tmul_mul_tmul, h_mul]
-        · intro x₁ x₂ h₁ h₂
-          rw [mul_add, map_add, map_add, mul_add, h₁, h₂]
-      · intro x₁ x₂ h₁ h₂
-        rw [add_mul, map_add, map_add, add_mul, h₁, h₂]
+  AlgHom.ofLinearMap f h_one <| f.map_mul_iff.2 <| by
+    -- these instances are needed by the statement of `ext`, but not by the current definition.
+    letI : Algebra R C := RestrictScalars.algebra R S C
+    letI : IsScalarTower R S C := RestrictScalars.isScalarTower R S C
```

Comment:
Should these be just plain `let`s?
```suggestion
    let _i1 : Algebra R C := RestrictScalars.algebra R S C
    let _i2 : IsScalarTower R S C := RestrictScalars.isScalarTower R S C
```
(Even after reading the tactic documentation I don't fully understand when I'll need to the inlining that `letI` provides.)

Reply:
This was a combination of lean 3 muscle memory, and an unwillingness to have to name these.


I suspect that there are still good reasons to use `letI` for instances as a general rule, but indeed it makes no difference here

Reply:
I'm going to leave these as `letI`s because it matches the mathportism that are already endemic to mathlib4; but as you saw I created https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/let.20vs.20letI.20for.20local.20instances/near/396246664 to revisit this.